### ISSUE TITLE
BatchFile use syntax?

### ISSUE THREAD

**Issue by tleylan** [Bluegrams/YoutubeDLSharp#56]
I am trying to download just thumbnails for a list of video Ids I have obtained in a search. I have it working on a thumbnail at a time but hoped that using the Batchfile option would mean I wouldn't be calling this method in a loop.

I was using RunVideoDownload but found RunWithOptions and for thumbnails that works every bit as well. But both take a url as the first argument. I don't understand the point of that if it is supposed to read the file and process the urls in that file. Maybe it is ignored but so far I haven't gotten the BatchFile approach to work.

Anything special I should know about?

I need the downloads converted to jpg as well and again it works on the single call solution. I would like to streamline the process a bit. I also don't see how the file can be renamed using the batch file approach. It may be a limitation of using batch files.

In a nutshell. I have an array of Video Ids. I want to download the thumbnails, have them save to folder using the video ID as the name, convert to jpg and ultimately resize them (smaller). It works now just not smoothly. And it seems to not download all the thumbnails in some cases. Though If call it download again with the same video Id it does. I don't know if yt-dlp is getting confused or overwhelmed or if I need to watch for something.
